src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -60,7 +62,15 @@ public PersonCard(Person person, int displayedIndex, Logic logic, MainWindow mai
         if (person.hasDefaultImage()) {
             image = new Image(person.getImagePath());
         } else {
-            image = new Image("file:" + person.getImagePath());
+            try {
+                if (ImageUtil.isValidImage(person.getImagePath())) {
+                    image = new Image("file:" + person.getImagePath());
+                } else {
+                    image = new Image(Person.getDefaultImagePath());
+                }
+            } catch (IOException io) {
```

Comment:
Good code quality with exception handling. It throws exception if the reader cannot detect a file